contrib/selinux/syslog_ng.sh

```diff
@@ -1,63 +1,272 @@
-#!/bin/sh -e
-
-DIRNAME=`dirname $0`
-cd $DIRNAME
-USAGE="$0 [ --update ]"
-if [ `id -u` != 0 ]; then
-	echo 'You must be root to run this script'
-	exit 1
-fi
-
-if [ ! -f /etc/selinux/config ]; then
-	echo 'SELinux not installed on this machine.'
-	exit 1
-fi
-
-. /etc/selinux/config
-
-if [ -z "$SELINUX" ] || [ "$SELINUX" = "disabled" ]; then
-	echo 'SELinux not configured.'
-	exit 1
-fi
-
-if [ ! -f /usr/share/selinux/devel/Makefile ]; then
-	echo 'selinux-policy package missing. please install it.'
-	exit 1
-fi
-
-if [ $# -eq 1 ]; then
-	if [ "$1" = "--update" ] ; then
-		time=`ls -l --time-style="+%x %X" syslog_ng.te | awk '{ printf "%s %s", $6, $7 }'`
-		rules=`ausearch --start $time -m avc --raw -se syslog_ng`
-		if [ x"$rules" != "x" ] ; then
-			echo "Found avc's to update policy with"
-			echo -e "$rules" | audit2allow -R
-			echo "Do you want these changes added to policy [y/n]?"
-			read ANS
-			if [ "$ANS" = "y" -o "$ANS" = "Y" ] ; then
-				echo "Updating policy"
-				echo -e "$rules" | audit2allow -R >> syslog_ng.te
-				# Fall though and rebuild policy
-			else
-				exit 0
-			fi
+#!/bin/bash
+
+EL_FC=
+EL_TE=
+OS_VERSION=
+INSTALL_PATH="/opt/syslog-ng"
+# ports 514/udp, 6514/udp and 6514/tcp are allowed by default
+# if you wish to add further ports, just add them to the end of the list
+SYSLOG_NG_TCP_PORTS="601"
+SYSLOG_NG_UDP_PORTS="601"
+TASK_SELECTED="install"
+INPUT=
+
+query_install_path() {
+	echo -n "Please enter your installation path for Syslog-ng PE: [${INSTALL_PATH}] "
+	read INPUT
+}
+
+check_dir() {
+	if [ -d "${1}" ]; then
+		return 0
+	else
+		echo "The directory you specified does not exist!"
+		return 1
```

Comment:
Different return values should be used on different errors. If someone runs it from a script, the chance is taken away from him or her to react accordingly based on the error.